tf-modules/s3-credentials/main.tf

```diff
@@ -0,0 +1,227 @@
+  }
+
+  environment {
+    variables = {
+      DISTRIBUTION_ENDPOINT          = var.external_api_endpoint
+      DISTRIBUTION_REDIRECT_ENDPOINT = "${var.external_api_endpoint}redirect"
+      public_buckets                 = join(",", var.public_buckets)
+      EARTHDATA_BASE_URL             = var.urs_url
+      EARTHDATA_CLIENT_ID            = var.urs_client_id
+      EARTHDATA_CLIENT_PASSWORD      = var.urs_client_password
+      AccessTokensTable              = aws_dynamodb_table.access_tokens.id
+      STSCredentialsLambda           = var.sts_credentials_lambda_function_arn
+    }
+  }
+  tags = var.tags
+}
+
+data "aws_region" "current" {}
+
+resource "aws_lambda_permission" "lambda_permission" {
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.s3_credentials.function_name
+  principal     = "apigateway.amazonaws.com"
+
+  # The /*/*/* part allows invocation from any stage, method and resource path
+  # within API Gateway REST API.
+  source_arn = "arn:aws:execute-api:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:${var.rest_api_id}/*/*/*"
+}
+
+# GET /s3credentials
+resource "aws_api_gateway_resource" "s3_credentials" {
+  rest_api_id = var.rest_api_id
+  parent_id = var.rest_api_root_resource_id
+  path_part   = "s3credentials"
+}
+
+resource "aws_api_gateway_method" "s3_credentials" {
+  rest_api_id   = var.rest_api_id
+  resource_id   = aws_api_gateway_resource.s3_credentials.id
+  http_method   = "GET"
+  authorization = "NONE"
+}
+
+resource "aws_api_gateway_integration" "s3_credentials" {
+  rest_api_id             = var.rest_api_id
+  resource_id             = aws_api_gateway_resource.s3_credentials.id
+  http_method             = aws_api_gateway_method.s3_credentials.http_method
+  integration_http_method = "POST"
+  type                    = "AWS_PROXY"
+  uri                     = aws_lambda_function.s3_credentials.invoke_arn
+}
+
+# GET /redirect
+resource "aws_api_gateway_resource" "s3_credentials_redirect" {
+  rest_api_id = var.rest_api_id
+  parent_id = var.rest_api_root_resource_id
+  path_part   = "redirect"
+}
+
+resource "aws_api_gateway_method" "s3_credentials_redirect" {
+  rest_api_id   = var.rest_api_id
+  resource_id   = aws_api_gateway_resource.s3_credentials_redirect.id
+  http_method   = "GET"
+  authorization = "NONE"
+}
+
+resource "aws_api_gateway_integration" "s3_credentials_redirect" {
+  rest_api_id             = var.rest_api_id
+  resource_id             = aws_api_gateway_resource.s3_credentials_redirect.id
+  http_method             = aws_api_gateway_method.s3_credentials_redirect.http_method
+  integration_http_method = "POST"
+  type                    = "AWS_PROXY"
+  uri                     = aws_lambda_function.s3_credentials.invoke_arn
+}
+
+# GET /s3credentialsREADME
+resource "aws_api_gateway_resource" "s3_credentials_readme" {
+  rest_api_id = var.tea_rest_api_id
+  parent_id   = var.tea_rest_api_root_resource_id
```

Comment:
I can't deploy this because it seems you missed a number of places when you were updating the variables to get rid of the `tea_` prefix.